src/Data/AbstractData.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Silktide\ProspectClient\Data;
+
+use ArrayObject;
+
+abstract class AbstractData extends ArrayObject
+{
+    public function set(string $key, $value): void
```

Comment:
Extending from the ArrayObject is generally a bad idea, as it gives you an object with a wide public API that you can't really do anything about. (Plus, argh, SPL!)

In the case of the uses of the AbstractData, I would argue that none of them are required and just increase the complexity for no gain.

In both cases it looks like the primary way they're used is as a regular array, then with a `__toString()` method that is being called to prep it for behaviour that is then used in the HttpClient wrapper.

I suspect it would make much more sense if in both cases you just had the Api take a standard array and apply your transformations inside the wrapper.